pkg/controller/che/che_controller.go

```diff
@@ -1308,22 +1151,21 @@ func createConsoleLink(isOpenShift4 bool, protocol string, instance *orgv1.CheCl
 // GetEndpointTlsCrt creates a test TLS route and gets it to extract certificate chain
 // There's an easier way which is to read tls secret in default (3.11) or openshift-ingress (4.0) namespace
 // which however requires extra privileges for operator service account
-func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, endpointUrl string) (certificate []byte, err error) {
-	testRoute := &routev1.Route{}
+func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, endpointUrl string, clusterAPI deploy.ClusterAPI) (certificate []byte, err error) {
 	var requestURL string
+	var routeStatus deploy.RouteProvisioningStatus
+
 	if len(endpointUrl) < 1 {
-		testRoute = deploy.NewTlsRoute(instance, "test", "test", 8080)
-		logrus.Infof("Creating a test route %s to extract router crt", testRoute.Name)
-		if err := r.CreateNewRoute(instance, testRoute); err != nil {
-			logrus.Errorf("Failed to create test route %s: %s", testRoute.Name, err)
-			return nil, err
-		}
-		// sometimes timing conditions apply, and host isn't available right away
-		if len(testRoute.Spec.Host) < 1 {
+		for wait := true; wait; {
+			routeStatus = deploy.SyncRouteToCluster(instance, "test", "test", 8080, clusterAPI)
+			if routeStatus.Err != nil {
+				return nil, routeStatus.Err
+			}
+			wait = !routeStatus.Continue || len(routeStatus.Route.Spec.Host) == 0
```

Comment:
Maybe here we should also try to do something non-blocking, returning a status as well ?

Reply:
I will try

Reply:
From the other hand creation of route it is a matter of couple of seconds.
And we need to delete test route after creation. So, the logic will be quite complicated. I would like to postpone these changes

Reply:
I understand. As you like.